packages/controller/src/index.ts

```diff
@@ -17,39 +13,52 @@ import {
   ResponseCodes,
   ConnectReply,
   ProbeReply,
-  Modal,
   ControllerOptions,
-  ControllerThemePresets,
-  ColorMode,
   PaymasterOptions,
-  // Prefund,
   ConnectError,
+  Profile,
+  IFrames,
 } from "./types";
-import { createModal } from "./modal";
-import { defaultPresets } from "./presets";
-import { NotReadyToConnect } from "./errors";
-import { KEYCHAIN_URL, RPC_SEPOLIA } from "./constants";
+import { KeychainIFrame, ProfileIFrame } from "./iframe";
+import { NotReadyToConnect, ProfileNotReady } from "./errors";
+import { RPC_SEPOLIA } from "./constants";
 
 class Controller {
-  private url: URL;
   private policies: Policy[];
   private paymaster?: PaymasterOptions;
-  private connection?: Connection<Keychain>;
-  private modal?: Modal;
   public keychain?: AsyncMethodReturns<Keychain>;
+  public profile?: AsyncMethodReturns<Profile>;
```

Comment:
I wonder if `controller.keychain` has to be public 🤔

Reply:
Yeah good point, I think both keychain and profile should be private